File: src/app/Service/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private _http: HttpClient) {}


  public getQuestionsOfQuiz(qid:any) {
    return this._http.get(`http://localhost:8081/Wellbeignatwork/question/quiz/all/${qid}`);
  }

  public getQuestionsOfQuizForTest(qid:any) {
    return this._http.get(`http://localhost:8081/Wellbeignatwork/question/quiz/${qid}`);
  }

  //add question
  public addQuestion(question:any) {
    return this._http.post(`http://localhost:8081/Wellbeignatwork/question/`, question);
  }
  //delete question
  public deleteQuestion(questionId:any) {
    return this._http.delete(`http://localhost:8081/Wellbeignatwork/question/${questionId}`);
  }

  //eval quiz
  public evalQuiz(questions:any) {
    return this._http.post(`http://localhost:8081/Wellbeignatwork/question/eval-quiz`, questions);
  }


}

File: src/main/java/com/wellbeignatwork/backend/service/Collaboration/QuestionService.java
package com.wellbeignatwork.backend.service.Collaboration;

import com.wellbeignatwork.backend.entity.Collaboration.QuestionCollaboration;
import com.wellbeignatwork.backend.entity.Collaboration.Quiz;

import java.util.Set;

public interface QuestionService {

	public QuestionCollaboration addQuestion(QuestionCollaboration question);

	public QuestionCollaboration updateQuestion(QuestionCollaboration question);

	public Set<QuestionCollaboration> getQuestions();

	public QuestionCollaboration getQuestion(Long questionId);

    public void deleteQuestion(Long quesId);

	public Set<QuestionCollaboration> getQuestionsOfQuiz(Quiz quiz);

	public QuestionCollaboration get(Long questionId);
}
